enhancements/security/x509-trust.md

```diff
@@ -0,0 +1,183 @@
+Cluster components may also perform other X.509 verification besides TLS requests.
+For example, user mail processing components may perform [S/MIME][smime] verification.
+This enhancement provides a framework for managing X.509 trust for these situations.
+
+Currently the default trust for CRI-O, the kubelet's cloud provider, and other node-namespace egress is the node's X.509 certificate authority (CA) store, which depends on the operating system ([Red Hat Enterprise Linux CoreOS (RHCOS)][rhcos], etc.) and any subsequent configuration ([the machine-config operator][machine-config-operator], etc.).
+The default trust for containers like the machine-API operator is that container's CA store, which depends on the base layer ([Red Hat Universal Base Image (UBI)][ubi], etc.).
+Many external components will have X.509 certificates signed by one of those default CAs, but this is not always the case.
+
+This enhancement provides a mechanism for configuring additional or alternative trust bundles for all cluster services when communicating with external components whose X.509 certificates are not signed by a default CA.
+
+## Motivation
+
+### Goals
+
+Administrators will have a single location to configure a default additional trust bundle to be used by all cluster components communicating with external components.
+They will also be able to specify whether this default is to be use in addition to, or instead of, the component's default trust bundle.
+
+Administrators will have per-component configuration locations to adjust the cluster-scoped default.
+As for the cluster-scoped default, they will be able to specify whether the per-component trust bundle is to be used in addition to, or instead of, the cluster-scoped default.
+
+### Non-Goals
+
+This enhancement does not address components making intra-cluster TLS requests.
+Trust for those requests is covered by [the Kubernetes API-server Certificates enhancement](../kube-apiserver/certificates) and [the Automated Service CA Rotation enhancement](../automated-service-ca-rotation.md).
+
+This enhancement does not address configuration for non-CA TLS settings like ciphers and allowed TLS versions.
+[The ingress TLS configuration enhancement](../kube-apiserver/tls-config.md) covers those settings for cluster-hosted services.
+Outgoing TLS probably needs similar configuration, but that configuration is decoupled from X.509 trust and should be addressed in a separate enhancement.
+
+## Proposal
+
+### Cluster-scoped Default Trust
+
+There will be a ConfigMap (FIXME: or should this be a new CRD under openshift/api?) named `default-ca-bundle` in the `openshift-config` namespace.
+The ConfigMap will have the following keys:
+
+* `ca-bundle.crt`, containing a PEM-encoded X.509 certificate bundle.
+* `composition`, configuring how the bundle in `ca-bundle.crt` is composed with the component's default trust store.
+    Valid values are:
```

Comment:
i don't think the components themselves should think about this, rather only care about the created-as-effect configmap for the most part... so wording it in that sense is more appropriate imo

Reply:
> i don't think the components themselves should think about this...

They need to think about it, because currently they all union the additional trust with their local default store, and that gives us no way to say "actually, forget about your default store, and just use this".  Right?

Reply:
> It may be worth expanding on why it's useful to discard the certs from the default trust store.

Done with e5d3ceb -> 47290b8.